app/models/bet.rb

```diff
@@ -2,4 +2,42 @@ class Bet < ApplicationRecord
   has_many :users, through: :user_bets
   has_many :user_bets
   has_many :options, through: :user_bets, source: :bet_option
+
+  def resolve
+    losers.each do |debtor|
+      debt = UserBet.find_by(user: debtor, bet: self).amount_bet
+      win_contributions.each do |contribution|
+        Debt.create(debtor: debtor,
+                    creditor: contribution[:creditor],
+                    amount:   debt * contribution[:percentage])
+      end
+    end
+  end
+
+  def win_option
+    options.find(&:winner)
+  end
+
+  def winner_total
+    UserBet.where(bet: self, bet_option: win_option).map(&:amount_bet).sum
+  end
+
+  def winners
+    UserBet.where(bet: self, bet_option: win_option).map do |ub|
+      User.find(ub.user_id)
+    end
+  end
+
+  def losers
+    UserBet.where(bet: self).where.not(bet_option: win_option).map do |ub|
+      User.find(ub.user_id)
+    end
+  end
+
+  def win_contributions
```

Comment:
Could be probably also be simplified with a scope on UserBet.

Returning an array of hashes seems a little weird to me here